contrib/exporters/awsflowlogs/mod/mangle_logstatus.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/flow"
+)
+
+type mangleLogStatus struct {
+	linkIDs map[string]true
+	flows   map[string]*flow.Flow
+}
+
+// Mangle log-status book keeping
+func (m *mangleLogStatus) Mangle(in []*flow.Flow) (out []*flow.Flow) {
+	now := common.UnixMillis(time.Now())
+	// OK: all flows
+	for _, flow := range in {
+		okFlow := *flow
+		okFlow.LogStatus = FlowLogStatus_OK
+		out = append(out, &okFlow)
+	}
+
+	// NODATA: when nothing was seen per intrerface in measurement window
+	linkIDs := make(map[string]true)
+	for _, flow := range in {
+		linkIDs[flow.Link.ID] = true
+	}
+
+	for key := range linkIDs {
+		if !m.linkIDs[key] {
+			noFlow := &flow.Flow{
+				Link: &FlowLayer{
+					ID: key,
+				},
+				Start:     now,
+				Last:      now,
+				LogStatus: FlowLogStatus_NODATA,
+			}
+			out = append(out, noFlow)
+		}
+
+	}
+
+	for _, flow := range in {
+		m.linkIDs[flow.Link.ID] = true
+	}
+
+	// SKIPDATA: flows were lost due to error or resource constraint
+	for _, flow := range in {
+		key := flow.TrackingID + "," + flow.NodeTID
+		if last, ok := m.flows[key]; ok {
+			if flow.LastUpdateMetadata != m.flows[key].Metadata {
+				skipFlow := &flow.Flow{
+					Link:           flow.Link,
+					Network:        flow.Network,
+					TransportLayer: flow.TransportLayer,
+					ICMPLayer:      flow.ICMPLayer,
+					Start:          now,
+					Last:           now,
+					TrackingID:     flow.TrackingID,
+					L3TrackingID:   flow.L3TrackingID,
+					NodeTID:        flow.NodeTID,
+					LogStatus:      FlowLogStatus_SKIPDATA,
```

Comment:
LogStatus - Status

Reply:
The mangle interface looks identical for what we do with Accept/Reject Action. Do you allow more than one mangler in a pipeline? Can we do both LogStatus and Accept/Reject with multiple instances of the same mechanism?

Reply:
the idea is to create a mangle named for example "logstatus+action" which applies both one after the other. (so instead of making the mangle configuration more complicated I simply will create a composite mangle.

Reply:
@eranra - its' not `Status` of the flow its' the status of reporting hence the AWS term `LogStatus`